libs/blocks/merch-card/merch-card.js

```diff
@@ -112,103 +203,156 @@ const decorateMerchCardLinkAnalytics = (el) => {
   });
 };
 
-const init = (el) => {
+const init = async (el) => {
+  const styles = [...el.classList];
+  const lastClass = styles[styles.length - 1];
```

Comment:
What if the class order does not have the variant name at last position?

Reply:
Authors can add the merch card name (not the variant name) as the last class.
If it is missing, it is fine.
The variant name is resolved using your podType function and it falls back to product by default.